corehq/apps/hqcase/api/get_list.py

```diff
@@ -9,28 +14,55 @@
 MAX_PAGE_SIZE = 5000
 
 
+def _to_boolean(val):
+    return val.lower() not in [''] + list(FALSE_STRINGS)
+
+
 def _to_int(val, param_name):
     try:
         return int(val)
     except ValueError:
         raise UserError(f"'{val}' is not a valid value for '{param_name}'")
 
 
+def _make_date_filter(date_filter, param):
+
+    def filter_fn(val):
+        try:
+            # If it's only a date, don't turn it into a datetime
+            val = datetime.date.fromisoformat(val)
+        except ValueError:
+            try:
+                val = parse(val)
+            except ValueError:
+                raise UserError(f"Cannot parse datetime '{val}'")
+        return date_filter(**{param: val})
+
+    return filter_fn
+
+
+def _to_date_filters(field, date_filter):
+    return [
+        (f'{field}.gt', _make_date_filter(date_filter, 'gt')),
+        (f'{field}.gte', _make_date_filter(date_filter, 'gte')),
+        (f'{field}.lte', _make_date_filter(date_filter, 'lte')),
+        (f'{field}.lt', _make_date_filter(date_filter, 'lt')),
```

Comment:
Should we offer `eq`?

The user could use, for example, `date_opened.gte=1878-02-19T00:00:00&date_opened.lt=1878-02-20T00:00:00`.

But `date_opened.eq=1878-02-19` might be nice.

Reply:
Oooh, I like that idea.  I might back log it for now though, as it would require a bit of figuring to work out.

Some thoughts:
Do we translate dates to midnight-to-midnight ranges like you mentioned?  What if the user passes in a datetime?  Do we figure out the degree of precision they specify and do a range on that or round to the nearest millisecond or something?  ES might handle this well already somehow (can you do a term query on a datetime field?) - we could make new filters for those or make it a standard arg for [date range filters](https://github.com/dimagi/commcare-hq/blob/2b8a63d094f99a8282fc27ff9e877d3eba2f59e4/corehq/apps/es/filters.py#L69-L74).

Reply:
Maybe a silly question, but what's the difference between the `.eq` and just  `date_opened=1878-02-19`?

Reply:
if you do modify this, I would do it somewhere around here instead, because this is how we do proper date queries with the case search index. 

https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/es/case_search.py#L208-L211

Reply:
and now having said that... I think you should just be able to use that `case_property_range_query` (or something that calls it, like: https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/es/case_search.py#L87-L92)

Reply:
I ended up implementing this in this PR.